i3pystatus/openvpn/__init__.py

```diff
@@ -0,0 +1,73 @@
+from i3pystatus import IntervalModule
+
+
+class Openvpn(IntervalModule):
+    """
+    Generic OpenVPN manager
+
+    The `backend` setting determines the backends to use.
+
+    Example for conffile backend:
+
+    .. code-block:: python
+
+        status.register('openvpn',
+            vpn_name='OpenVPN',
+            backend=conffile.File(
+                conf_file='/home/user/ovpn.conf'))
+
+    Example for systemd backend:
+
+    .. code-block:: python
+
+        status.register('openvpn',
+            vpn_name='OpenVPN',
+            backend=systemd.Systemd)
+    """
+
+    settings = (
+        ("backend", "OpenVPN management backend (instances of ``i3pystatus.openvpn.xxx.zzz``, e.g. :py:class:`.file.File`)"),
+        ("format", "Format string"),
+        ("color_up", "VPN is up"),
+        ("color_down", "VPN is down"),
+        ("vpn_name", "Name of this VPN connection"),
+        ("label", "Label for this VPN connection"),
+        ("status_down", "Symbol to display when down"),
+        ("status_up", "Symbol to display when up"),
+    )
+
+    color_up = "#00ff00"
+    color_down = "#FF0000"
+    status_up = '▲'
+    status_down = '▼'
+    vpn_name = 'OpenVPN'
+    label = ''
+    format = "{vpn_name} {status}"
+    backend = None
+    connected = False
+    on_leftclick = ['toggle_connection']
+
+    def toggle_connection(self):
+        self.backend.toggle_connection()
+
+    def run(self):
+        if 'systemd.Systemd' in str(self.backend):
```

Comment:
Not really a fan of this, seems like there must be a better way. Also this code executes `backend.init()` each time `run` is called which is a bit weird, usually you expect `init` to be executed once on initialization.